docker/errors.go

```diff
@@ -0,0 +1,32 @@
+package docker
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// ErrV1NotSupported is returned when we're trying to talk to a
+	// docker V1 registry.
+	ErrV1NotSupported = errors.New("can't talk to a V1 docker registry")
+	// ErrUnauthorizedForCredentials is returned when the status code returned is 401
+	ErrUnauthorizedForCredentials = errors.New("unable to retrieve auth token: invalid username/password")
+	// ErrTooManyRequests is returned when the status code returned is 429
+	ErrTooManyRequests = errors.New("too many request to registry")
```

Comment:
`github.com/pkg/errors.New()` records a stack trace that leads back here.  I'm guessing that the standard library's `errors.New()` would work better, perhaps using `github.com/pkg/errors.WithStack()` to wrap them before returning them in `httpResponseToError()`.

Reply:
Sounds like a nice improvement!